### PR TITLE
feat: topic management

### PR DESCRIPTION
**Pull request by fdelbrayelle**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/fdelbrayelle/generator-jhipster-kafka/actions) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/fdelbrayelle/generator-jhipster-kafka/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/app/prompts.js [hunk 1]
```diff
@@ -308,10 +328,27 @@ function askForUnitaryEntityOperations(generator, done) {
...
+            message: 'For which topic?',
+            choices: topicsChoices(generator, previousConfiguration(generator)),
+            default: constants.DEFAULT_TOPIC
+        },
+        {
+            when: response => response.topic === constants.CUSTOM_TOPIC,
+            type: 'input',
+            name: 'topicName',
+            message: 'What is the topic name?',
+            validate: input => (_.isEmpty(input) ? 'You have to choose a topic name' : true)
```
Comment (monxxi) [inline generators/app/prompts.js:351]: add a pattern validation to avoid some string issue (like spaces or special character),For example, i can put a topic name like "ok go" and it works, but i think it shouldn't no?
KafkaTopic

File: generators/app/prompts.js [hunk 2]
```diff
@@ -380,3 +427,17 @@ function pushComponentsByEntity(generator, answers, entity) {
...
 }
+
+function pushTopicName(generator, answers) {
+    const name = generator.props.currentEntity === constants.NO_ENTITY ? generator.props.currentPrefix : generator.props.currentEntity;
+
+    if (answers.topic) {
+        if (answers.topic === constants.CUSTOM_TOPIC) {
+            generator.props.topics.push({ key: name, value: answers.topicName });
+        } else if (answers.topic === constants.DEFAULT_TOPIC) {
+            generator.props.topics.push({ key: name, value: utils.topicNamingFormat(generator, name) });
```
Comment (monxxi) [inline generators/app/prompts.js:438]: when i use the incremental mode with the default topic choice, it generates a wrong topic name. (undefined instead of appName in snakeCase)
WrongTopicName

File: generators/app/prompts.js [hunk 3]
```diff
@@ -56,6 +56,26 @@ function componentChoices() {
...
+        value: constants.DEFAULT_TOPIC
+    });
+    topicsChoices.push({
+        name: 'Custom topic name',
+        value: constants.CUSTOM_TOPIC
+    });
+
+    if (previousConfiguration && previousConfiguration.topics) {
+        Object.entries(previousConfiguration.topics).forEach(topic => {
+            topicsChoices.push({ name: topic[1], value: topic[0] });
```
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply generators/app/prompts.js:72]: I disagree with that suggestion because value should be the key so `topic[0]`.

File: generators/app/prompts.js (hunk 1 above)
Comment (fdelbrayelle) [reply generators/app/prompts.js:351]: You're right!

File: generators/app/templates/src/main/resources/application-kafka.yml.ejs [hunk 4]
```diff
@@ -31,10 +30,19 @@ _%>
...
 <%_ } _%>
 <%_ } _%>
+<%_ if (components.includes('consumer') || components.includes('producer')) { _%>
+  topic:
+<%_ for (entity of entities) {
+        if (entity === 'no_entity' && componentsPrefixes.length > 0 && componentsPrefixes[0]) {
+            entity = componentsPrefixes[0];
+        }
+_%>
+    <%= _.camelCase(entity) _%>: queuing.<%= snakeCaseBaseName %>.<%= _.snakeCase(entity) %>
```
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply generators/app/templates/src/main/resources/application-kafka.yml.ejs:45]: Not clear, can you rephrase please?

File: generators/app/prompts.js [hunk 5]
```diff
@@ -56,6 +56,26 @@ function componentChoices() {
...
+    topicsChoices.push({
+        name: 'Default topic name following this convention: message_type.application_type.entity_name',
+        value: constants.DEFAULT_TOPIC
+    });
+    topicsChoices.push({
+        name: 'Custom topic name',
+        value: constants.CUSTOM_TOPIC
+    });
+
+    if (previousConfiguration && previousConfiguration.topics) {
```
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply generators/app/prompts.js:70]: Yes because you may want to consume or produce in same topics.

File: generators/app/templates/src/main/resources/application-kafka.yml.ejs (hunk 4 above)
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply generators/app/templates/src/main/resources/application-kafka.yml.ejs:45]: Ok as seen together, I'll add a topic choice (default or custom) on each entity in bigbang mode.
